prediction_market_agent/agents/microchain_agent/nft_treasury_game/app_nft_treasury_game.py

```diff
@@ -64,13 +69,32 @@ def prompt_table_handler(identifier: AgentIdentifier) -> PromptTableHandler:
     return PromptTableHandler.from_agent_identifier(identifier)
 
 
+@st.dialog("Send message to agent")
+def send_message_via_wallet(
+    recipient: ChecksumAddress, message: str, amount_to_send: float
+) -> None:
+    wallet_component(
+        recipient=recipient,
+        amount_in_ether=f"{amount_to_send:.10f}",  # formatting number as 0.0001000 instead of scientific notation
+        data=message,
+    )
+
+
 def send_message_part(nft_agent: type[DeployableAgentNFTGameAbstract]) -> None:
     message = st.text_area("Write a message to the agent")
-
+    amount_to_send = st.number_input(
+        "Amount to send to agent (xDAI)",
+        min_value=OMEN_TINY_BET_AMOUNT,
+        step=OMEN_TINY_BET_AMOUNT,
+        format="%0.5f",
+    )
+    message_compressed = HexBytes(compress_message(message)).hex() if message else ""
```

Comment:
```suggestion
    message_compressed = HexBytes(compress_message(message)).hex()
```

Any harm in simplifying this?